yt_dlp/extractor/svt.py

```diff
@@ -252,50 +257,29 @@ def _real_extract(self, url):
         mobj = self._match_valid_url(url)
         video_id = mobj.group('id')
         svt_id = mobj.group('svt_id') or mobj.group('modal_id')
-
         if svt_id:
             return self._extract_by_video_id(svt_id)
 
         webpage = self._download_webpage(url, video_id)
 
-        data = self._parse_json(
-            self._search_regex(
-                self._SVTPLAY_RE, webpage, 'embedded data', default='{}',
-                group='json'),
-            video_id, fatal=False)
-
-        thumbnail = self._og_search_thumbnail(webpage)
-
-        if data:
-            video_info = try_get(
-                data, lambda x: x['context']['dispatcher']['stores']['VideoTitlePageStore']['data']['video'],
-                dict)
-            if video_info:
-                info_dict = self._extract_video(video_info, video_id)
-                info_dict.update({
-                    'title': data['context']['dispatcher']['stores']['MetaStore']['title'],
-                    'thumbnail': thumbnail,
-                })
-                return info_dict
-
-            svt_id = try_get(
-                data, lambda x: x['statistics']['dataLake']['content']['id'],
-                str)
-
-        if not svt_id:
-            nextjs_data = self._search_nextjs_data(webpage, video_id, fatal=False)
-            svt_id = traverse_obj(nextjs_data, (
-                'props', 'urqlState', ..., 'data', {json.loads}, 'detailsPageByPath',
-                'video', 'svtId', {str}), get_all=False)
-
+        data = traverse_obj(self._search_nextjs_data(webpage, video_id), (
+            'props', 'urqlState', ..., 'data', {json.loads},
+            'detailsPageByPath', {dict}, any, {require('video data')}))
+        details = traverse_obj(data, (
+            'modules', lambda _, v: v['details']['smartStart']['item']['videos'],
+            'details', {dict}, any))
```

Comment:
as pointed out by dirkf, the `{dict}` check is redundant
```suggestion
            'modules', lambda _, v: v['details']['smartStart']['item']['videos'], 'details', any))
```